File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
class Solution {
    public int[] solution(int n, String[] words) {
        int[] answer = new int[2];
        
        int a = 0;
        int b = 0;

        loop1: for(int i = 0;i<words.length;i++){
            for(int j = 0; j<i; j++){
                if(words[i].equals(words[j])){
                    a = (i%n) + 1;
                    b = (i/n) + 1;
                    break loop1;
                }
            }
            if(i > 0 &&
              words[i-1].charAt(words[i-1].length() - 1) != 
               words[i].charAt(0)
              ){
                a = (i%n) + 1;
                b = (i/n) + 1;
                break;
            }
        }
        
        answer[0] = a;
        answer[1] = b;

        return answer;
    }
}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.cpp
#include <bits/stdc++.h>


using namespace std;

int ret, a,b;
map<string, int> m;
char c;

vector<int> solution(int n, vector<string> words) {
    vector<int> answer;
    
    for (string word : words) {
        ret++;
        
        if (ret > 1 && c != word[0]) {
            a = ret % n;
            if (a == 0) a = n;
            b = ret / n;
            if (ret % n != 0) b++;
            answer.push_back(a);
            answer.push_back(b);
            break;
        }
        
        if (m[word] != 0) {
            a = ret % n;
            if (a == 0) a = n;
            b = ret / n;
            if (ret % n != 0) b++;
            answer.push_back(a);
            answer.push_back(b);
            break;
        }
            
        c = word[word.size() - 1];  
        m[word]++;
        
    }
    if (answer.size() == 0) {
        answer.push_back(0);
        answer.push_back(0);
    }

    return answer;
}